src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioAnalyzer.cs

```diff
@@ -24,22 +23,29 @@ internal sealed class VisualStudioAnalyzer : IDisposable
         private readonly IAnalyzerAssemblyLoader _loader;
         private readonly string _language;
 
+        private readonly object _gate;
+
         private AnalyzerReference _analyzerReference;
-        private List<DiagnosticData> _analyzerLoadErrors;
+        private ImmutableArray<DiagnosticData> _analyzerLoadErrors;
```

Comment:
Please add a comment that modifications of this must be done under _gate.

Reply:
sure.